hyperspy/io_plugins/emd.py

```diff
@@ -317,21 +325,29 @@ def save_to_emd(self, filename='datacollection.emd'):
         emd_file.attrs['version_major'] = ver_maj
         emd_file.attrs['version_minor'] = ver_min
         # Write user:
-        user_group = emd_file.require_group('user')
         for key, value in self.user.items():
-            user_group.attrs[key] = value
+            if not (value==''):
+                if 'user' not in emd_file:
```

Comment:
I believe this check is unnecessary, as the `require_group` does exactly this - checks if the group exists, and then either returns the reference if so, or creates it otherwise.

I could be wrong though